app/config/Switches.scala

```diff
@@ -0,0 +1,103 @@
+package config
+
+import akka.agent.Agent
+import com.amazonaws.services.s3.model.{GetObjectRequest, ObjectMetadata, PutObjectRequest}
+import com.amazonaws.util.StringInputStream
+import org.quartz._
+import org.quartz.impl.StdSchedulerFactory
+import play.api.Logger
+import play.api.libs.json.{Format, Json}
+import play.json.extra.Variants
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.io.Source
+
+object Switches {
+  def allSwitches: Map[String, SwitchState] = agent.get()
+  private val agent = Agent[Map[String, SwitchState]](Map.empty)
+  private val bucket = "login-gutools-config"
+  private val scheduler = StdSchedulerFactory.getDefaultScheduler
+  val loginConfig = LoginConfig.loginConfig(AWS.eC2Client)
+  val fileName = s"${loginConfig.stage.toUpperCase}/switches.json"
+
+  class SwitchJob extends Job() {
+    override def execute(context: JobExecutionContext): Unit = refresh()
+  }
+  private val job = JobBuilder.newJob(classOf[SwitchJob])
+    .withIdentity("refresh-switches-gu-login-tools")
+    .build()
+
+  def setEmergencySwitch(state: SwitchState) {
+    val name = "emergency"
+    val newStates = allSwitches + (name -> state)
+
+    val json = Json.toJson(newStates)
+    val jsonString = Json.stringify(json)
+    val metaData = new ObjectMetadata()
+    metaData.setContentLength(jsonString.getBytes("UTF-8").length)
+    val request = new PutObjectRequest(bucket, fileName, new StringInputStream(jsonString), metaData)
+
+    try {
+      AWS.s3Client.putObject(request)
+    } catch {
+      case e: Exception => {
+        Logger.error(s"Unable to update switch $name ${state.name}", e)
+        throw e
```

Comment:
Rather than catching and re-throwing errors and then catching errors in the controllers as well, would it be better to have this return something?